src/zebra_puzzles/clue_removal.py

```diff
@@ -0,0 +1,186 @@
+"""Module for removing redundant clues."""
+
+import numpy as np
+
+from zebra_puzzles.zebra_solver import solver
+
+
+def remove_redundant_clues_part1(
+    new_clue: str,
+    chosen_clues: list[str],
+    clue_par: tuple[str, list[int], np.ndarray],
+    clue_pars: list,
+    clue_type: str,
+    clue_types: list[str],
+    prioritise_old_clues: bool = False,
+) -> tuple[bool, list[int]]:
+    """Use simple rules to check if a suggested clue is redundant.
+
+    This is to avoid using the solver for every clue suggestion and thereby speed up the clue selection process.
+
+    NOTE: More checks could be added e.g. "same_object" and "not_same_object" with 1 identical attribute and secondary attributes of the same category.
+    NOTE: Consider adapting for non-unique attributes
+    TODO: Combine checks for fewer loops
+
+    Args:
+        new_clue: The clue to check as a string.
+        chosen_clues: Chosen clues for the zebra puzzle as a list of strings.
+        clue_par: Clue parameters for the new clue.
```

Comment:
The type hint states that it's a tuple with three elements - could you state what these are here?

Reply:
Perfect, thanks